crates/api_common/src/utils.rs

```diff
@@ -973,12 +976,8 @@ pub fn generate_followers_url(actor_id: &DbUrl) -> Result<DbUrl, ParseError> {
   Ok(Url::parse(&format!("{actor_id}/followers"))?.into())
 }
 
-pub fn generate_inbox_url(actor_id: &DbUrl) -> Result<DbUrl, ParseError> {
-  Ok(Url::parse(&format!("{actor_id}/inbox"))?.into())
-}
-
-pub fn generate_shared_inbox_url(settings: &Settings) -> LemmyResult<DbUrl> {
-  let url = format!("{}/inbox", settings.get_protocol_and_hostname());
+pub fn generate_inbox_url() -> LemmyResult<DbUrl> {
+  let url = format!("{}/inbox", SETTINGS.get_protocol_and_hostname());
```

Comment:
We gotta be wary with globals. If it were me I'd still want the settings to get passed in from the context, as in L980 of the last one.

Reply:
`context.settings()` is just an indirect, more verbose way to do the exact same thing, because it also uses the global.

Reply:
I won't block it for this one to get it merged, but we really should make sure every function has all the params it needs, even if those are globals behind the scenes.